### DISCUSSION TITLE
Update all versions in package.json

### DISCUSSION THREAD

**Discussion by mikethecalamity** [renovatebot/renovate#28917] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Is there a configuration to force all versions in `package.json` to be updated, even those with `^`?

In the logs I see a new version was found and put in the `package-lock.json`, but not each individual `package.json`.

```json
{
    "datasource": "npm",
    "depName": "pg",
    "displayPending": "",
    "fixedVersion": "8.7.3",
    "currentVersion": "8.7.3",
    "currentValue": "^8.5.1",
    "newValue": "^8.5.1",
    "newVersion": "8.11.5",
    "packageFile": "js/packages/myproject/package.json",
    "updateType": "minor",
    "packageName": "pg"
}
```

I guess I'm looking to update the `newValue`?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You will need to use `rangeStrategy: pin` for this.
Option: https://docs.renovatebot.com/configuration-options/#rangestrategy
Related Reading: https://docs.renovatebot.com/dependency-pinning/

**Reply by rarkins** [reply]
Alternatively `rangeStrategy=bump` if you want to keep the `^` in place

**Reply by Niceplace** [reply]
Is `rangeStrategy: pin` supposed to convert a range to a fixed version in `package.json` ?
Assuming that this is set within a `packageRules` entry, which I understand should override any other `rangeStrategy` defined elsewhere, right ?

Basically I am looking for a way for renovate to convert my ranges to pinned dependencies and I don't understand how to make this work. 

Example: `"typescript": "^5.0.0"` -> `"typescript": "5.0.0"`

**Reply by rarkins** [reply]
Please start a new discussion

**Comment by mikethecalamity** [comment]
Awesome thanks!
